Types fail to build for ponder.schema.ts

### Version

0.10.23

### Current behavior

I am trying to export a `.d.ts` file for my ponder.schema.ts so that I can create a ponder client for this ponder instance inside of a separate project. Is this supported?

Here is a link to my [tsconfig.json](https://github.com/ethereum-optimism/ecosystem/blob/f9cc75ea5acc152fde422d5fa5dad42c7ca2e701/apps/ponder-interop/tsconfig.json)

The types for ponder.schema.ts fail to build with this error on each `onchaintable`:
```
error TS2742: The inferred type of 'sentMessages' cannot be named without a reference to '.pnpm/drizzle-orm@0.41.0_@electric-sql+pglite@0.2.13_@opentelemetry+api@1.9.0_kysely@0.26.3_pg@8.14.1/node_modules/drizzle-orm/pg-core'. This is likely not portable. A type annotation is necessary.
```

Here is my build script: https://github.com/ethereum-optimism/ecosystem/blob/f9cc75ea5acc152fde422d5fa5dad42c7ca2e701/apps/ponder-interop/package.json#L7

### Expected behavior

The expected behavior is that the type generation should succeed so that the schema is able to exported from my project and used by another project.

### Steps to reproduce

This can be reproed by running `pnpm build` from this repository: https://github.com/ethereum-optimism/ecosystem/tree/f9cc75ea5acc152fde422d5fa5dad42c7ca2e701/apps/ponder-interop. 

### Link to repository

https://github.com/ethereum-optimism/ecosystem/tree/f9cc75ea5acc152fde422d5fa5dad42c7ca2e701/apps/ponder-interop

### Anything else?

_No response_

--- a/.changeset/rare-moons-fry.md
+++ b/.changeset/rare-moons-fry.md
@@ -1,0 +1,5 @@
+---
+"ponder": patch
+---
+
+Exported more types to fix error in `ponder.schema.ts`: `error TS2742: The inferred type of '[table]' cannot be named without a reference`.

--- a/packages/core/src/drizzle/onchain.ts
+++ b/packages/core/src/drizzle/onchain.ts
@@ -213,7 +213,7 @@ export const onchainTable = <
 
 export const isPgEnumSym = Symbol.for("drizzle:isPgEnum");
 
-export interface OnchainEnum<TValues extends [string, ...string[]]> {
+export type OnchainEnum<TValues extends [string, ...string[]]> = {
   (): PgEnumColumnBuilderInitial<"", TValues>;
   <TName extends string>(
     name: TName,
@@ -227,12 +227,12 @@ export interface OnchainEnum<TValues extends [string, ...string[]]> {
   readonly schema: string | undefined;
   /** @internal */
   [isPgEnumSym]: true;
-}
+} & { [onchain]: true };
 
 export const onchainEnum = <U extends string, T extends Readonly<[U, ...U[]]>>(
   enumName: string,
   values: T | Writable<T>,
-): OnchainEnum<Writable<T>> & { [onchain]: true } => {
+): OnchainEnum<Writable<T>> => {
   const schema = globalThis?.PONDER_NAMESPACE_BUILD?.schema;
   const e = pgEnumWithSchema(enumName, values, schema);
 
@@ -344,6 +344,7 @@ function pgEnumWithSchema<U extends string, T extends Readonly<[U, ...U[]]>>(
       enumValues: values,
       schema,
       [isPgEnumSym]: true,
+      [onchain]: true,
     } as const,
   );
 

--- a/packages/core/src/drizzle/onchain.ts
+++ b/packages/core/src/drizzle/onchain.ts
@@ -111,7 +111,7 @@ export type OnchainTable<
   enableRLS: () => Omit<OnchainTable<T>, "enableRLS">;
 };
 
-type BuildExtraConfigColumns<
+export type BuildExtraConfigColumns<
   columns extends Record<string, ColumnBuilderBase>,
 > = {
   [key in keyof columns]: ExtraConfigColumn & {
@@ -119,7 +119,7 @@ type BuildExtraConfigColumns<
   };
 };
 
-type PgColumnsBuilders = Omit<
+export type PgColumnsBuilders = Omit<
   _PgColumnsBuilders,
   "bigint" | "serial" | "smallserial" | "bigserial"
 > & {

--- a/packages/core/src/index.ts
+++ b/packages/core/src/index.ts
@@ -32,6 +32,14 @@ export {
   bigint,
 } from "@/drizzle/onchain.js";
 
+export type {
+  OnchainTable,
+  BuildExtraConfigColumns,
+  PgColumnsBuilders,
+  PrimaryKeyBuilder,
+  OnchainEnum,
+} from "@/drizzle/onchain.js";
+
 export type { ReadonlyDrizzle } from "@/types/db.js";
 
 export { client } from "@/client/index.js";

